sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs

```diff
@@ -258,9 +291,112 @@ protected async Task ProcessOneMessageWithinScopeAsync(ServiceBusReceivedMessage
             }
         }
 
-        private static ICollection<ServiceBusReceivedMessage> GetProcessedMessages(EventArgs args) =>
-            args is ProcessMessageEventArgs processMessageEventArgs ? processMessageEventArgs.Messages.Keys
-                : ((ProcessSessionMessageEventArgs)args).Messages.Keys;
+        private async Task ProcessMessages(
+            IReadOnlyList<ServiceBusReceivedMessage> triggerMessages,
+            CancellationToken cancellationToken)
+        {
+            ServiceBusErrorSource errorSource = ServiceBusErrorSource.Receive;
+            EventArgs args = ConstructEventArgs(triggerMessages, cancellationToken);
+
+            try
+            {
+                errorSource = ServiceBusErrorSource.ProcessMessageCallback;
+
+                // TODO resolve for batch
```

Comment:
is this still a TODO?

Reply:
I resolved and removed the todo.
Created helper methods to log messages in the list.